### PR TITLE
indexeddb.getAttachments(): reduce selects from N+1

### PR DESCRIPTION
**Pull request by alxndrsn**
This one weird trick (probably) improves perf by quite a bit.

### CODE REVIEW

**Comment by alxndrsn** [comment]
Here's a couple of new perf tests to go with it:

### current `master`
```js
  ✓ basic-attachments
       { median: 10, numIterations: 1000 }
  ✓ many-attachments-base64
       { median: 1133, numIterations: 100 }
  ✓ many-attachments-binary
       { median: 2259, numIterations: 100 }
```

### this branch
```js
  ✓ basic-attachments
       { median: 10, numIterations: 1000 }
  ✓ many-attachments-base64
       { median: 59.5, numIterations: 100 }
  ✓ many-attachments-binary
       { median: 15, numIterations: 100 }
```

**Review by garethbowen — APPROVED** [review]
Nice quick win, thanks!

File: packages/node_modules/pouchdb-adapter-indexeddb/src/getAttachment.js [hunk 1]
```diff
@@ -2,9 +2,16 @@
...
 
-import { DOC_STORE } from './util';
+function getAttachment(txn, docId, attachId, _, opts, cb) {
+  if (txn.error) {
+    return cb(txn.error);
+  }
+
+  const doc = opts.metadata;
+  const rev = doc.revs[opts.rev || doc.rev].data;
+  const digest = rev._attachments[attachId].digest;
```
Comment (SourceR85) [inline packages/node_modules/pouchdb-adapter-indexeddb/src/getAttachment.js:12]: Just a thought: Why did you need to lookup the digest from doc.revs?
The fourth parameter `_` already has the corresponding attachment-data (refer [core/adapter.js:704](https://github.com/pouchdb/pouchdb/blob/c5e103e32f0ce4e8e510726aaa867d3479665615/packages/node_modules/pouchdb-core/src/adapter.js#L704))

Comment (alxndrsn) [reply packages/node_modules/pouchdb-adapter-indexeddb/src/getAttachment.js:12]: I just copied the existing code :grin: I'll give that fourth parameter a try, thanks!

Comment (alxndrsn) [reply packages/node_modules/pouchdb-adapter-indexeddb/src/getAttachment.js:12]: Testing at https://github.com/alxndrsn/pouchdb/actions/runs/6460505912

Comment (alxndrsn) [reply packages/node_modules/pouchdb-adapter-indexeddb/src/getAttachment.js:12]: @SourceR85 great suggestion, thanks :+1: and it looks like it works :grin:
